Remove old UI and webserver

bbovenzi opened the pull request:
![excited-fuego](https://github.com/user-attachments/assets/56b90dbd-48a0-4358-92ed-167477815c1f)



---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

bbovenzi committed:
First run at deleting the legacy UI and webserver

jedcunningham commented on .pre-commit-config.yaml:
```diff
@@ -883,32 +860,6 @@ repos:
-        files: ^providers/fab/.*/www/
-        entry: ./scripts/ci/pre_commit/compile_fab_assets.py
-        pass_filenames: false
-        additional_dependencies: [ 'yarn@1.22.21' ]
```
This one needs to stay.

vincbeck replied:
Good catch! Yes please :) Keep it

bbovenzi replied:
Fixed

jedcunningham commented on airflow/www/app.py:
Why are we keeping any of this?

vincbeck replied:
+1

bbovenzi replied:
Deleted.

jedcunningham commented on providers/databricks/src/airflow/providers/databricks/plugins/databricks_workflow.py:
```diff
@@ -387,7 +387,7 @@ def get_link(
 
 
-class RepairDatabricksTasks(AirflowBaseView, LoggingMixin):
+class RepairDatabricksTasks(BaseView, LoggingMixin):
```
Hmm, should this extend the FAB AirflowBaseView, or be moved to a normal flask blueprint instead? Either way, some compat work with AF2 needs to be done here I think.

jedcunningham commented on providers/edge/src/airflow/providers/edge/plugins/edge_executor_plugin.py:
```diff
@@ -36,8 +36,7 @@
 from airflow.www.auth import has_access_view
-from airflow.www.constants import SWAGGER_BUNDLE, SWAGGER_ENABLED
-from airflow.www.extensions.init_views import _CustomErrorRequestBodyValidator, _LazyResolver
+from airflow.providers.fab.www.extensions.init_views import _CustomErrorRequestBodyValidator, _LazyResolver
```
These imports should just move, unchanged, into `_get_airflow_2_api_endpoint` instead. That's already only run on AF2.

jscheffl replied:
Need to double check that this does not break compatability with Airflow 2 - I am not sure whether plugin is loaded in providers test. Once CI is green let me check this manually please.

bbovenzi replied:
Moved.

jscheffl replied:
Looks better now!

vincbeck commented on .pre-commit-config.yaml:
```diff
@@ -616,12 +606,10 @@ repos:
           ^airflow/models/baseoperator.py$|
           ^airflow/operators/__init__.py$|
           ^providers/common/sql/tests/provider_tests/common/sql/operators/test_sql_execute.py$|
-          ^providers/fab/src/airflow/providers/fab/www/static/css/bootstrap-theme.css$|
```
We need to keep it

bbovenzi replied:
I put it back

vincbeck commented on .pre-commit-config.yaml:
```diff
@@ -632,7 +620,6 @@ repos:
           ^providers/apache/spark/src/airflow/providers/apache/spark/operators/|
           ^providers/exasol/src/airflow/providers/exasol/hooks/exasol.py$|
           ^providers/fab/src/airflow/providers/fab/auth_manager/security_manager/|
-          ^providers/fab/src/airflow/providers/fab/www/static/css/bootstrap-theme.css$|
```
We need to keep it

jscheffl commented on airflow/ui/CONTRIBUTING.md:
```diff
@@ -28,7 +28,7 @@ Manually:
 - Have the `dev-mode` environment variable enabled
 - Run `pnpm install && pnpm dev`
-- Note: Make sure to access the UI via the Airflow localhost port (8080 or 28080) and not the vite port (5173)
+- Note: Make sure to access the UI via the Airflow localhost port (9091 or 29091) and not the vite port (5173)
```
Maybe not in this PR but in a follow-up I'd propose to change fastAPI to 8080 and the external exposure in breeze to 28080 as a full replacement.

jedcunningham replied:
Yep, agree. It should probably become "webserver" too.

jscheffl commented on dev/breeze/src/airflow_breeze/global_constants.py:
```diff
@@ -54,7 +54,8 @@
 NONE_BACKEND = "none"
-ALLOWED_BACKENDS = [SQLITE_BACKEND, MYSQL_BACKEND, POSTGRES_BACKEND, NONE_BACKEND]
+ALLOWED_BACKENDS = [SQLITE_BACKEND,
+                    MYSQL_BACKEND, POSTGRES_BACKEND, NONE_BACKEND]
```
This re-formatting seems to be odd - different line length in ruff?

jscheffl commented on dev/breeze/src/airflow_breeze/utils/run_tests.py:
```diff
@@ -220,12 +222,18 @@ def get_excluded_provider_args(python_version: str) -> list[str]:
-    all_named_test_folders.extend(TEST_GROUP_TO_TEST_FOLDERS[GroupOfTests.HELM])
-    all_named_test_folders.extend(TEST_GROUP_TO_TEST_FOLDERS[GroupOfTests.INTEGRATION_CORE])
-    all_named_test_folders.extend(TEST_GROUP_TO_TEST_FOLDERS[GroupOfTests.INTEGRATION_PROVIDERS])
+    all_named_test_folders = list(chain.from_iterable(
```
This also seems like a mis-configured ruff?

jscheffl commented on generated/provider_dependencies.json:
```diff
@@ -535,7 +535,9 @@
     ],
-    "cross-providers-deps": [],
+    "cross-providers-deps": [
+      "fab"
```
I assume the cross-dependency must be added in provider yaml as well?

jedcunningham commented on chart/values.yaml:
```diff
@@ -295,27 +295,6 @@ airflowPodAnnotations: {}
 
-# `airflow_local_settings` file as a string (templated).
-airflowLocalSettings: |-
-  {{- if semverCompare ">=2.2.0" .Values.airflowVersion }}
```
Can't throw this away. We should do a semver compare < 3 also.

jedcunningham replied:
Done!

ashb committed:
fixup! First run at deleting the legacy UI and webserver

ashb committed:
fixup! First run at deleting the legacy UI and webserver

jedcunningham committed:
Fix static checks

jedcunningham committed:
Remove airflow.www.security_manager

jedcunningham committed:
Fix chart default airflowlocalsettings

This will need to be adjusted again when we have UIAlert support in AF3,
but this will do for now :)

jedcunningham committed:
FAB is now a optional dep for the amazon provider

jedcunningham committed:
Stop testing api_connexion
